api/app/weather_models/process_grib.py

```diff
@@ -148,6 +189,47 @@ def yield_data_for_stations(self, raster_band: gdal.Dataset):
 
             yield (points, values)
 
+    def yield_uv_wind_data_for_stations(self, u_raster_band: gdal.Dataset, v_raster_band: gdal.Dataset, variable: str):
+        """ Given a list of stations and 2 gdal datasets (one for u-component of wind, one for v-component
+        of wind), yield relevant data 
+        """
+        for station in self.stations:
+            longitude = station.long
+            latitude = station.lat
+
+            x_coordinate, y_coordinate = calculate_raster_coordinate(
+                longitude, latitude, self.padf_transform, self.geo_to_raster_transformer)
+
+            if 0 <= x_coordinate < u_raster_band.XSize and 0 <= x_coordinate < v_raster_band.XSize and\
+                    0 <= y_coordinate < u_raster_band.YSize and 0 <= y_coordinate < v_raster_band.YSize:
+                u_points, u_values = get_surrounding_grid(u_raster_band, x_coordinate, y_coordinate)
+                v_points, v_values = get_surrounding_grid(v_raster_band, x_coordinate, y_coordinate)
+
+                assert u_points == v_points
+
+                zipped_uv_values = list(zip(u_values, v_values))
+
+                if variable == 'wdir_tgl_10':
+                    yield self.get_wind_dir_values(u_points, zipped_uv_values)
+                elif variable == 'wind_tgl_10':
+                    yield self.get_wind_speed_values(u_points, zipped_uv_values)
+            else:
+                logger.warning('coordinate not in u/v wind rasters - %s', station)
+
+    def get_wind_dir_values(self, u_points: List[int], zipped_uv_values):
+        """ Get calculated wind direction values for list of points and zipped u,v values """
+        wind_dir_values = []
+        for u, v in zipped_uv_values:
+            wind_dir_values.append(calculate_wind_dir_from_u_v(u, v))
+        return (u_points, wind_dir_values)
+
+    def get_wind_speed_values(self, u_points: List[int], zipped_uv_values):
```

Comment:
Is it possible to write unit tests for this function since it's new and self contained?

Reply:
Yeah I see now thanks, sounds like tests for `yield_uv_wind_data_for_stations` would be valuable